pyiron_workflow/storage.py

```diff
@@ -297,11 +297,15 @@ def available_backends(
     """
 
     standard_backends = {"pickle": PickleStorage}
-    backend = standard_backends.get(backend, PickleStorage)() if isinstance(backend, str) else backend
-
-    if backend is not None:
-        yield backend
+    backend_instance = (
+        standard_backends.get(backend, PickleStorage)()
+        if isinstance(backend, str)
+        else backend
+    )
+
+    if backend_instance is not None:
+        yield backend_instance
         if only_requested:
             return
 
-    yield from (v() for v in standard_backends.values() if not isinstance(backend, v))
+    yield from (v() for k, v in standard_backends.items() if k != backend)
```

Comment:
Well, you reverted it back to a semi-broken state. `k` is of type `str` and backend is of type `Literal["pickle"] | StorageInterface | None`. If you did not implement a custom `__eq__` operator this is broken for `StorageInterface`.

Reply:
Sounds good to me, go ahead and stack it on